kong/plugins/aws-lambda/handler.lua

```diff
@@ -19,20 +31,62 @@ function AWSLambdaHandler:new()
   AWSLambdaHandler.super.new(self, "aws-lambda")
 end
 
-local function retrieve_parameters()
-  ngx_req_read_body()
-
-  return utils.table_merge(ngx_req_get_uri_args(), public_utils.get_body_args())
-end
-
 function AWSLambdaHandler:access(conf)
   AWSLambdaHandler.super.access(self)
 
-  local bodyJson = cjson.encode(retrieve_parameters())
+  local upstream_body = new_tab(0, 6)
+
+  do
+    ngx_req_read_body()
+
+    local body_args, err_code, body_raw = public_utils.get_body_info()
+    if err_code == public_utils.req_body_errors.unknown_ct then
+      -- don't know what this body MIME type is, base64 it just in case
+      body_raw = ngx_encode_base64(body_raw)
```

Comment:
initially i was thinking im not entirely sure about this approach. what about content types that are already base64 encoded. perhaps a flag to encode unknown types vs. passing them through. but not a blocker, and we start to run into edge case territory here

Reply:
@akh00 Flag added! ;)